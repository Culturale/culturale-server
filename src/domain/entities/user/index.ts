export * from './user';
export * from './user.interface';
export * from './user.schema';
