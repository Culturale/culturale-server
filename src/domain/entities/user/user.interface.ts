export interface IUser {
  id: string;
  email: string;
  username: string;
}
