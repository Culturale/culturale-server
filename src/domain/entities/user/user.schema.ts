import { Schema, model } from "mongoose";

import type { IUser } from "./user.interface";

const UserSchema = new Schema({
  email: { required: true, type: String },
  username: { required: true, type: String, unique: true },
});

const UserModel = model<IUser>("User", UserSchema);

export { UserModel };
