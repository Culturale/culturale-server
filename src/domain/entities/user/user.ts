import type { IUser } from './user.interface';

export class User implements IUser {
    public username: string;
    public name: string;
    public password: string;
    public email: string;
    public profilePicture: string;
    public phoneNumber: string;
    public usertype: string;

    constructor(username: string, name: string, password: string, email: string, profilePicture: string, phoneNumber: string, usertype: string) {
        this.username = username;
        this.name = name;
        this.password = password;
        this.email = email;
        this.profilePicture = profilePicture;
        this.phoneNumber = phoneNumber;
        this.usertype = usertype;
    }
}
