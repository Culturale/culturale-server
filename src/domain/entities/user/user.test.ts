import { User } from './user';
import type { IUser } from './user.interface';

describe('User Entity', function () {
  let instance: IUser;

  beforeEach(function () {
    instance = new User('test-username', 'test-name','test-password','test-email','test-profilePicture','test-phone','test-type');

  });

  it('can be created', function () {
    expect(instance).toMatchSnapshot();
  });
});
