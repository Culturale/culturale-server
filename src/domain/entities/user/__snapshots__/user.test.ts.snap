// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Entity can be created 1`] = `
User {
  "email": "test-email",
  "name": "test-name",
  "password": "test-password",
  "phoneNumber": "test-phone",
  "profilePicture": "test-profilePicture",
  "username": "test-username",
  "usertype": "test-type",
}
`;
